diff --git a/submit/submit.cc b/submit/submit.cc
index e295b9b98..46814210d 100644
--- a/submit/submit.cc
+++ b/submit/submit.cc
@@ -391,6 +391,10 @@ int main(int argc, char **argv)
 	langid = stringtolower(langid);
 	for(i=0; i<languages.size(); i++) {
 		for(j=0; j<languages[i].extensions.size(); j++) {
+			if ( languages[i].id == "python2" &&
+				 stringtolower(languages[i].extensions[j]) == "py" ) {
+				continue;
+			}
 			if ( stringtolower(languages[i].extensions[j]) == langid ) {
 				mylanguage = languages[i];
 				goto lang_found;
diff --git a/webapp/src/Controller/API/ContestController.php b/webapp/src/Controller/API/ContestController.php
index 259cabeb7..0cb6b3334 100644
--- a/webapp/src/Controller/API/ContestController.php
+++ b/webapp/src/Controller/API/ContestController.php
@@ -428,6 +428,10 @@ class ContestController extends AbstractRestController
                     if (!$strict) {
                         $result['time'] = Utils::absTime($event->getEventtime());
                     }
+                    // NWERC 2019 'hack': do not expose company teams
+                    if ($result['type'] === 'groups' && $result['op'] !== 'delete' && $result['data']['name'] === 'Companies') {
+                        $result['data']['hidden'] = true;
+                    }
                     echo json_encode($result, JSON_PRESERVE_ZERO_FRACTION | JSON_UNESCAPED_SLASHES) . "\n";
                     ob_flush();
                     flush();
diff --git a/webapp/templates/partials/javascript_language_detect.html.twig b/webapp/templates/partials/javascript_language_detect.html.twig
index 477a1064f..7ee8680e6 100644
--- a/webapp/templates/partials/javascript_language_detect.html.twig
+++ b/webapp/templates/partials/javascript_language_detect.html.twig
@@ -4,8 +4,10 @@
         switch (ext) {
                 {% for language in submission_languages %}
                 {% for extension in language.extensions %}
+                {% if language.langid != 'py2' or extension != 'py' %}
             case '{{ extension }}':
                 return '{{ language.langid }}';
+                {% endif %}
                 {% endfor %}
                 {% endfor %}
             default:
diff --git a/webapp/templates/partials/scoreboard.html.twig b/webapp/templates/partials/scoreboard.html.twig
index b874d9c4a..aa75a93dc 100644
--- a/webapp/templates/partials/scoreboard.html.twig
+++ b/webapp/templates/partials/scoreboard.html.twig
@@ -134,6 +134,7 @@
 
     <p id="lastmod">
         Last Update: {{ null | printtime('%a %d %b %Y %T %Z') }}<br/>
-        using <a href="https://www.domjudge.org/" target="_top">DOMjudge</a>
+        using <a href="https://www.domjudge.org/" target="_top">DOMjudge</a><br />
+        University logos from <a href="https://open.kattis.com/">Kattis</a>
     </p>
 {% endif %}
diff --git a/webapp/templates/team/menu.html.twig b/webapp/templates/team/menu.html.twig
index c64f7ac21..05ce53ba5 100644
--- a/webapp/templates/team/menu.html.twig
+++ b/webapp/templates/team/menu.html.twig
@@ -21,6 +21,9 @@
                     <a class="nav-link disabled"><i class="fas fa-book-open"></i> Problemset</a>
                 {% endif %}
             </li>
+            <a class="nav-link" href="https://docs/" target="_blank">
+                <i class="fas fa-file-alt"></i> Docs
+            </a>
             {% if have_printing %}
                 <li class="nav-item {% if current_route == 'team_print' %}active{% endif %}">
                     <a class="nav-link" href="{{ path('team_print') }}"><i class="fas fa-file-alt"></i> Print</a>
